src/stripeJavaLanguageClient/utils.ts

```diff
@@ -0,0 +1,665 @@
+  Starting = 'Starting',
+  Started = 'Started',
+  Error = 'Error',
+  Stopping = 'Stopping',
+}
+
+export enum ServerMode {
+  STANDARD = 'Standard',
+  LIGHTWEIGHT = 'LightWeight',
+  HYBRID = 'Hybrid',
+}
+
+export namespace StatusNotification {
+  export const type = new NotificationType<StatusReport>('language/status');
+}
+
+export interface FindLinksParams {
+  type: string;
+  position: TextDocumentPositionParams;
+}
+
+export interface LinkLocation extends Location {
+  displayName: string;
+  kind: string;
+}
+
+export namespace FindLinks {
+  export const type = new RequestType<FindLinksParams, LinkLocation[], void>('java/findLinks');
+}
+
+export function getJavaConfiguration(): WorkspaceConfiguration {
+  return workspace.getConfiguration('java');
+}
+
+export async function ensureNoBuildToolConflicts(
+  context: ExtensionContext,
+  outputChannel: OutputChannel,
+): Promise<boolean> {
+  const isMavenEnabled: boolean =
+    getJavaConfiguration().get<boolean>('import.maven.enabled') || false;
+  const isGradleEnabled: boolean =
+    getJavaConfiguration().get<boolean>('import.gradle.enabled') || false;
+  if (isMavenEnabled && isGradleEnabled) {
+    const activeBuildTool: string | undefined = context.workspaceState.get(ACTIVE_BUILD_TOOL_STATE);
+    if (!activeBuildTool) {
+      if (!(await hasBuildToolConflicts())) {
+        return true;
+      }
+      outputChannel.appendLine('Build tool conflicts are detected in workspace.');
+      return false;
+    }
+  }
+
+  return true;
+}
+
+async function hasBuildToolConflicts(): Promise<boolean> {
+  const projectConfigurationUris: Uri[] = await getBuildFilesInWorkspace();
+  const projectConfigurationFsPaths: string[] = projectConfigurationUris.map((uri) => uri.fsPath);
+  const eclipseDirectories = getDirectoriesByBuildFile(projectConfigurationFsPaths, [], '.project');
+  // ignore the folders that already has .project file (already imported before)
+  const gradleDirectories = getDirectoriesByBuildFile(
+    projectConfigurationFsPaths,
+    eclipseDirectories,
+    '.gradle',
+  );
+  const gradleDirectoriesKts = getDirectoriesByBuildFile(
+    projectConfigurationFsPaths,
+    eclipseDirectories,
+    '.gradle.kts',
+  );
+  gradleDirectories.concat(gradleDirectoriesKts);
+  const mavenDirectories = getDirectoriesByBuildFile(
+    projectConfigurationFsPaths,
+    eclipseDirectories,
+    'pom.xml',
+  );
+  return gradleDirectories.some((gradleDir) => {
+    return mavenDirectories.includes(gradleDir);
+  });
+}
+
+async function getBuildFilesInWorkspace(): Promise<Uri[]> {
+  const buildFiles: Uri[] = [];
+  const inclusionFilePatterns: string[] = getBuildFilePatterns();
+  inclusionFilePatterns.push('**/.project');
+  const inclusionFolderPatterns: string[] = getInclusionPatternsFromNegatedExclusion();
+  // Since VS Code API does not support put negated exclusion pattern in findFiles(),
+  // here we first parse the negated exclusion to inclusion and do the search.
```

Comment:
When is this used? Is this for cases where we want to exclude everything in `/path/a` unless it's `/path/a/b`?

Reply:
this it to handle the java configuration settings. ie:
```
"java.import.exclusions": [
     "**/node_modules/**",
    "**/.metadata/**",
    "**/archetype-resources/**",
    "**/META-INF/maven/**",
    "**/Frontend/**",
],
```
if user puts `!` in front of any exclusion, it means `include`